src/gmail/Module.md

```diff
@@ -82,62 +46,167 @@ gmail:GmailConfiguration gmailConfig = {
 
 gmail:Client gmailClient = new (gmailConfig);
 ```
-The `sendMessage` remote function sends an email. `MessageRequest` object which contains all the data is required
-to send an email. The `userId` represents the authenticated user and can be a Gmail address or ‘me’ 
-(the currently authenticated user).
+
+## Feature Overview
+
+### Working with Emails
+
+The `wso2/gmail` module contains operations to send emails in Text and HTML formats with attachments and inline images. It supports searching and reading emails in Gmail using Gmail filters. The module also supports trashing, untrashing, deleting, and modifying emails.
+
+The following operations are possible using the Gmail Connector
+
+| Operation     | Description                                         |
+|---------------|-----------------------------------------------------|
+| listMessages  | Obtains list of emails that satisfy a search filter |
+| readMessage   | Read email                                          |
+| sendMessage   | Send email. Can be sent with attachments            |
+| modifyMessage | Modify email within the inbox                       |
+| deleteMessage | Delete email                                        |
+
+### - Retrieve List of messages
+
+A list of messages can be obtained using the `listMessages()` remote function.
+
+__Parameters__
+
+| Parameter Name | Data Type              | Description                                                                                    |
+|----------------|------------------------|------------------------------------------------------------------------------------------------|
+| userId         | string                 | The user's email address. The special value me can be used to indicate the authenticated user. |
+| filter         | gmail: MsgSearchFilter | OPTIONAL. MsgSearchFilter with optional query parameters to search messages.                   |
+
+
+To retrieve a specific set of Emails from the Inbox, a Search Filter needs to be defined. This is done using `gmail:MsgSearchFilter`. This contains the rules of the Search Filter in JSON format.
+
+```ballerina
+    gmail:MsgSearchFilter searchFilter = {includeSpamTrash: false, labelIds: [inboxLabel]};
```

Comment:
Shall we remove spaces and align with line 88

Reply:
Done